gc/base/MemoryPoolAddressOrderedList.cpp

```diff
@@ -621,22 +665,25 @@ MM_MemoryPoolAddressOrderedList::internalAllocateTLH(MM_EnvironmentBase *env, ui
 	}
 #endif /* OMR_GC_CONCURRENT_SWEEP */
 
-	if (doesNeedAlignment(env, freeEntry)) {
-		freeEntry = doFreeEntryAlignmentUpTo(env, freeEntry);
+	if (doesNeedCardAlignment(env, freeEntry)) {
+		freeEntry = doFreeEntryCardAlignmentUpTo(env, freeEntry);
 		if (NULL == freeEntry) {
 			goto retry;
 		}
 	}
 
-	/* Consume the bytes and set the return pointer values */
 	freeEntrySize = freeEntry->getSize();
-	Assert_MM_true(freeEntrySize >= _minimumFreeEntrySize);
-	consumedSize = (maximumSizeInBytesRequired > freeEntrySize) ? freeEntrySize : maximumSizeInBytesRequired;
+
 	_largeObjectAllocateStats->decrementFreeEntrySizeClassStats(freeEntrySize);
```

Comment:
LargeObjectStats are not used where CardAlignment is used (Balanced, the only one), so we don't a real problem now, but if ever both are needed, I suspect this is too late to do the decrement, since the card alignement can recycle the whole abandon the whole chunk.

@LinHu2016, do you agree - decrementFreeEntrySizeClassStats should be just before Card Alignment logic?

Reply:
@amicic Yes, I agree that decrementFreeEntrySizeClassStats should be just before getConsumedSizeForTLH(), since getConsumedSizeForTLH() could abandonHeapChunk  without decrementFreeEntrySizeClassStats and retry.
(LargeObjectStats is not used for balanced GC right now, CardAlignment is only for balanced GC).